src/formatDropdownValue.js

```diff
@@ -19,8 +19,10 @@ export default function formatDropdownValue(options) {
 
   const name = hit._highlightResult.locale_names[0].value;
   city = city ? hit._highlightResult.city[0].value : undefined;
-  administrative = administrative ? hit._highlightResult.administrative[0].value : undefined;
-  country = country ? hit._highlightResult.country.value : undefined;
+  administrative = administrative && hit._highlightResult.administrative ?
+    hit._highlightResult.administrative[0].value : administrative;
```

Comment:
if administrative is undefined we get undefined, otherwise the non highlighted administrative